grails-app/controllers/streama/VideoController.groovy

```diff
@@ -124,6 +124,8 @@ class VideoController {
     def file = uploadService.upload(request)
 
     if (file != null) {
+      file.isDefault = (videoInstance.videoFiles.isEmpty() && fileService.allowedVideoFormats.contains(file.extension)) ||
+        (videoInstance.getSubtitles().isEmpty() && fileService.allowedSubtitleFormats.contains(file.extension))
```

Comment:
you keep using a combination of 
`videoInstance.videoFiles.isEmpty() && fileService.allowedVideoFormats.contains(file.extension) 
`
and 

`videoInstance.getSubtitles().isEmpty() && fileService.allowedSubtitleFormats.contains(file.extension)`

in a lot places here, maybe it makes sense to extract this as instance methods on the video class?